### ISSUE TITLE
Responsive: PrimeFlex 3.0 Upgrade

### ISSUE THREAD

**Issue by melloware** [primefaces/primefaces#7598]
**Description**
PrimeFlex 3.0 was released and contains some breaking changes we would need to update in our components and our GridLayoutUtils.java

Migration Guide: https://www.primefaces.org/primeflex/migration

cc @christophs78 

TODOs:

- [x] Showcase
   - [x] all XHTMLs (use PrimeFaces CLI migration tool) 
   - [x] PrimeFlex section (documentation as part of XHTMLs)
- [x] Documentation
   - [x] Migration guide
   - [x] formlayout.md
   - [x] panelgrid.md
- [x] PrimeFaces (Core)
   - [x] PanelGrid
   - [x] GridLayoutUtils

**Comment by christophs78** [comment]
Don´t know yet wether i like all this class-renaming or not.
At least this should allow developers to include PrimeFlex 2 and PrimeFlex 3 without walking through the whole application and replacing css-classes.
Maybe someone want´s to write a migration-tool which walks through all XHTMLs and does this replacement automatically.

**Comment by melloware** [comment]
That is not a bad idea.  We were discussing something similar.

**Comment by christophs78** [comment]
We may provide such a PrimeFlex-Migration-CLI-Tool as new PrimeFaces-module or as part of PF Extensions.
Better to be done once than each costumer on it´s own.

**Comment by FlipWarthog** [comment]
@melloware Also this condition in PanelGridRenderer will need to be updated:
```java
if (!columnClass.contains("p-md-") && !columnClass.contains("p-col-")) {
   columnClass = columnClass + " " + GridLayoutUtils.getFlexColumnClass(columns);
}
```

**Comment by christophs78** [comment]
Regarding PrimeFlex-Migration-CLI-Tool the first step seems to be creating a file (JSON/XML/CSV) which contains a mapping for all PrimeFlex 2.0 - classes to PrimeFlex 3.0 - classes. I´ll try to start a PR for this.

**Comment by christophs78** [comment]
Do we still wan´t to do this for 11.0.0 or do we postpone this to 12.0.0?
When we wan´t to do this for 11.0.0 we would not to start early. Modify PrimeFaces Core, migrate Showcase incl. PrimeFlex - documentation.
From my personal view i would prefer a stable 11.0.0 release and do this afterwards.

**Comment by melloware** [comment]
I am OK waiting until 12.0 on this one.

**Comment by jepsar** [comment]
Seems like https://www.primefaces.org/primeblocks-jsf is using PrimeFlex 3... See the free samples: https://www.primefaces.org/primeblocks-jsf/free.xhtml

**Comment by buunsal** [comment]
I updated the showcase for PrimeFaces 10: https://www.primefaces.org/showcase/index.xhtml 
I will also add these changes to master before 11 release. PrimeFlex 2 docs will not change in the menu. We added PrimeFlex 3 link as an extra. We also do not intend to remove support for PrimeFlex 2 for PrimeFaces 11. But we will add support for the necessary components for PrimeFlex 3 as well, if needed. We are working on it.

Also In our other ui libraries, we support PrimeFlex 2 and 3 simultaneously in new versions. Anyone should be able to use whatever they want.
